src/System.IO.Pipelines/tests/PipeReaderStreamTests.nonnetstandard.cs

```diff
@@ -209,6 +243,41 @@ public void AsStreamReturnsSameInstance()
             Assert.Same(stream, pipeReader.AsStream());
         }
 
+        [Fact]
+        public async Task PipeWriterStreamProducesToConsumingPipeReaderStream()
```

Comment:
This is an interesting test. Any specific reason for it?

Reply:
Mainly because we have some approximations of this elsewhere in corefx tests, and I'd like to be able to substitute in Pipe rather than the manual implementations used there, e.g. see the VirtualNetwork type (and associated VirtualNetworkStream) that's used in the System.Net.Security tests.  The general building blocks are covered by other tests, e.g. that the reader wakes up when the writer completes, but it seemed desirable to have a higher-level test to help validate the producer/consumer via streams use case.  I'm not wed to it, though; if you think it's out-of-place, I can delete it.

Reply:
I think it's a good stress test, just wondered because it's not exactly connected to the change.

Reply:
One of the key pieces here is that the reader wakes up / returns 0 when the writer completes, so the using on the writer stream is one of the more relevant pieces.